### PR TITLE
Refactor docker container port select

### PR DESCRIPTION
**Pull request by madlabman**
Docker client accept port binding using START_PORT-END_PORT definition

### CODE REVIEW

File: src/adcm_pytest_plugin/docker_utils.py [hunk 1]
```diff
@@ -51,6 +53,7 @@ def _port_is_free(ip, port) -> bool:
     return True
 
 
+@deprecated()
```
Comment (dgusakov) [inline src/adcm_pytest_plugin/docker_utils.py:56]: These methods are private, so you can remove them at all

File: src/adcm_pytest_plugin/docker_utils.py [hunk 2]
```diff
@@ -296,17 +305,19 @@ def adcm_container(
         Return ADCM container and bind port.
         """
         for _ in range(0, CONTAINER_START_RETRY_COUNT):
-            port = _find_random_port(ip)
+            api_port = "8000/tcp"
```
Comment (Maunty) [inline src/adcm_pytest_plugin/docker_utils.py:308]: In this case i don`t we any need in for try except construction.

**Review by Maunty — CHANGES_REQUESTED** [review]

**Review by dgusakov — APPROVED** [review]

File: src/adcm_pytest_plugin/docker_utils.py (hunk 2 above)
Comment (madlabman) [reply src/adcm_pytest_plugin/docker_utils.py]: @Maunty, thank you, done

Comment (Maunty) [reply src/adcm_pytest_plugin/docker_utils.py]: @madlabman `class RetryCountExceeded` may be removed too but it is backward compatibility broke. Consult with @dgusakov .

**Review by Maunty — DISMISSED** [review]

File: src/adcm_pytest_plugin/docker_utils.py [hunk 3]
```diff
@@ -295,32 +283,17 @@ def adcm_container(
...
-                    "failed: port is already allocated" in err.explanation
-                    or "bind: address already in use" in err.explanation  # noqa: W503
-                ):
-                    # such error excepting leaves created container and there is
-                    # no way to clean it other than from docker library
-                    pass
-                else:
-                    raise err
-        else:
-            raise RetryCountExceeded(
```
Comment (dgusakov) [inline src/adcm_pytest_plugin/docker_utils.py:322]: Remove class definition in module above

File: src/adcm_pytest_plugin/docker_utils.py [hunk 4]
```diff
@@ -43,22 +41,6 @@ class RetryCountExceeded(Exception):
...
-        return False
-    return True
-
-
-def _find_random_port(ip) -> int:
-    for _ in range(0, 20):
-        port = random.randint(MIN_DOCKER_PORT, MAX_DOCKER_PORT)
-        if _port_is_free(ip, port):
-            return port
-    raise UnableToBind("There is no free port for Docker after 20 tries.")
```
Comment (dgusakov) [inline src/adcm_pytest_plugin/docker_utils.py:59]: Same here

**Review by dgusakov — CHANGES_REQUESTED** [review]

File: src/adcm_pytest_plugin/docker_utils.py (hunk 2 above)
Comment (madlabman) [reply src/adcm_pytest_plugin/docker_utils.py]: Removed

**Review by dgusakov — APPROVED** [review]

**Review by Maunty — APPROVED** [review]

**Pull request merged**